src/open_inwoner/haalcentraal/signals.py

```diff
@@ -19,8 +18,6 @@ def on_bsn_change(instance, **kwargs):
         and instance.is_prepopulated is False
         and instance.login_type == LoginTypeChoices.digid
     ):
-        system_action(
-            "Retrieving data from haal centraal based on BSN", content_object=instance
-        )
+        logger.info("Retrieving data for %s from haal centraal based on BSN", instance)
```

Comment:
I noticed the original created a TimelineLog while the change is a regular logger.

Reply:
I don't know but I think it could be important to have a TimelineLog entry for this functionality. 

Maybe the code can be re-arranged to work? I don't know if post-save would work because we modify the same instance. 

And at some point we should centralize/refactor the User signals (now or in the future): there are a few save signals floating around doing this kind of thing and it is hard to follow.

Reply:
Turns out it works after all. The relevant tests were failing at some point during the upgrade but are passing now.

Reply:
@Bartvaderkin Well that was too quick. I can get the haacentraal tests to pass, but not the DigiD tests. In fact, the combination of registration/login flow and logging itself is broken in its current form. When you try to register with a new BSN, the attempt to create a timeline log results in an error. I'll revert to the previous state (replace database logs with regular logging), however feel free to take a look.

Reply:
> However, I had to change to asserted log count in one of the tests

Practical learning experience :wink: 

If you want to test TimelineLog's it work better to not rely on ad-hoc queries and counts on TimelineLog model but check if the exact log(s) you expect is there.

Lot's of signals and things generate logs so the count in the tests is not reliable, because you don't know what other unexpected logs it also counts. And it obviously breaks if some other feature that logs is added and the count changes (bonus fun if a change drops one and adds another). 

This is why we have the AssertTimelineLogMixin that looks for the exact text and parameters. And yes there are old tests that do counts and it is an unreliable mess as you're now experiencing; we have no idea if they even tested correctly or not, and there are mystery counts floating around.

Anyway, if it had database logs before we'd expect them to remain after a refactor. We shouldn't just decide to not do that because it is inconvenient.

The idea of the TimelineLog is that we have a timeline in the admin and with some traceability and triage options if we have issues or complaints. It is not the same if we have to grab extra from some random log location. 

I would update the old tests and get rid of count based testing and do exact checks to verify the old tests were even correct at all, and move from there.

Reply:
It makes sense to clean up the logging tests that do rely on order and counts.

However, the test for haalcentraal signals does not fail because it relies on "ad-hoc queries and counts on TimelineLog model"; there are simply no logs created. I don't understand why/how the test could previously pass, since the function is triggered by a `pre-save` signal, but the log object to be created is related to the user object, which hasn't been saved at the time of logging.